OpenRA.Mods.Cnc/Activities/LayMines.cs

```diff
@@ -68,7 +68,7 @@ public override bool Tick(Actor self)
 
 			if ((minefield == null || minefield.Contains(self.Location)) && CanLayMine(self, self.Location))
 			{
-				if (rearmableInfo != null && ammoPools.Any(p => p.Info.Name == minelayer.Info.AmmoPoolName && !p.HasAmmo))
+				if (rearmableInfo != null && ammoPools.Any(p => p.CurrentAmmoCount >= minelayer.Info.AmmoUsage))
```

Comment:
This check looks inverted: We only need to go reloading if there is not any pool that has enough ammo.
```suggestion
				if (rearmableInfo != null && !ammoPools.Any(p => p.CurrentAmmoCount >= minelayer.Info.AmmoUsage))
```

Reply:
This should also change to work if no `rearmableInfo` is present.
```
if (!ammoPools.Any(p => p.CurrentAmmoCount >= minelayer.Info.AmmoUsage))
{
	if (rearmableInfo == null)
		<cancel the activity here? and then return>

	<current code for rearmableInfo != null>
}
```

Reply:
Ahh yep, misread what it was doing. Have put those fixes in.